airflow/providers/amazon/aws/hooks/quicksight.py

```diff
@@ -0,0 +1,152 @@
+                    target_state={"COMPLETED"},
+                    check_interval=check_interval,
+                    max_ingestion_time=max_ingestion_time,
+                )
+            return create_ingestion_response
+        except Exception as general_error:
+            self.log.error("Failed to run QuickSight create_ingestion API, error: %s", general_error)
+            raise
+
+    def get_status(self, aws_account_id: str, data_set_id: str, ingestion_id: str):
+        """
+        Get the current status of QuickSight Create Ingestion API.
+
+        :param aws_account_id: An AWS Account ID
+        :param data_set_id: QuickSight Data Set ID
+        :param ingestion_id: QuickSight Ingestion ID
+        :return: An QuickSight Ingestion Status
+        :rtype: str
+        """
+        try:
+            describe_ingestion_response = self.get_conn().describe_ingestion(
+                AwsAccountId=aws_account_id, DataSetId=data_set_id, IngestionId=ingestion_id
+            )
+            return describe_ingestion_response["Ingestion"]["IngestionStatus"]
+        except KeyError:
+            raise AirflowException("Could not get status of the QuickSight Ingestion")
+        except ClientError:
+            raise AirflowException("AWS request failed, check logs for more info")
+
+    def wait_for_state(
+        self,
+        aws_account_id: str,
+        data_set_id: str,
+        ingestion_id: str,
+        target_state: set,
+        check_interval: int,
+        max_ingestion_time: Optional[int] = None,
+    ):
+        """
+        Check status of a QuickSight Create Ingestion API
+
+        :param aws_account_id: An AWS Account ID
+        :param data_set_id: QuickSight Data Set ID
+        :param ingestion_id: QuickSight Ingestion ID
+        :param target_state: Describes the QuickSight Job's Target State
+        :param check_interval: the time interval in seconds which the operator
+            will check the status of QuickSight Ingestion
+        :param max_ingestion_time: the maximum ingestion time in seconds. QuickSight API if
+         run longer than this will fail. Setting this to None implies no timeout.
+        :return: response of describe_ingestion call after Ingestion is is done
+        """
+
+        sec = 0
+        status = self.get_status(aws_account_id, data_set_id, ingestion_id)
+        while status in self.non_terminal_states and status != target_state:
+            self.log.info("Current status is %s", status)
+            time.sleep(check_interval)
+            sec += check_interval
+            if status in self.failed_states:
+                raise AirflowException("QuickSight SPICE ingestion failed")
+            if status == "CANCELLED":
+                raise AirflowException("QuickSight SPICE ingestion cancelled")
+            if max_ingestion_time and sec > max_ingestion_time:
+                raise AirflowException(f"QuickSight Ingestion took more than {max_ingestion_time} seconds")
```

Comment:
I don't think we need `max_ingestion_time`
Operators have `execution_timeout` :
https://github.com/apache/airflow/blob/6126c4e40f36f76d294cfd72f84cb83d97d622a5/airflow/models/baseoperator.py#L722

which takes care of the need to fail a task if it didn't finish on specified timedelta.

Reply:
Thanks for the details ! I have made the changes. Please let me know if it looks fine.